packages/fx-core/src/core/index.ts

```diff
@@ -1271,7 +1284,10 @@ export class FxCore implements v3.ICore {
     projectSettings: ProjectSettings,
     inputs: Inputs
   ): Promise<Result<Void, FxError>> {
-    const appName = projectSettings.appName;
+    let appName = projectSettings.appName;
+    if (targetEnvName === environmentManager.getLocalEnvName()) {
+      appName = appName + "-local-debug";
```

Comment:
appName with suffix may exceed the manifest short name length limit of 30.

Suggest:
getLocalAppName() from "src/plugins/resource/appstudio/utils"

Reply:
Thanks for the advice, will update in next PR.